opal/mca/btl/openib/btl_openib_component.c

```diff
@@ -2547,19 +2512,14 @@ btl_openib_component_init(int *num_btl_modules,
     *num_btl_modules = 0;
     num_devs = 0;
 
-#if BTL_OPENIB_MALLOC_HOOKS_ENABLED
+#if MEMORY_LINUX_MALLOC_ALIGN_ENABLED
     /* If we got this far, then setup the memory alloc hook (because
        we're most likely going to be using this component). The hook
        is to be set up as early as possible in this function since we
-       want most of the allocated resources be aligned.*/
-    if (mca_btl_openib_component.use_memalign > 0 &&
-        (opal_mem_hooks_support_level() &
-            (OPAL_MEMORY_FREE_SUPPORT | OPAL_MEMORY_CHUNK_SUPPORT)) != 0) {
-        mca_btl_openib_component.previous_malloc_hook = __malloc_hook;
-        __malloc_hook = btl_openib_malloc_hook;
-        malloc_hook_set = true;
-    }
-#endif
+       want most of the allocated resources be aligned.
+     */
+    opal_memory_linux_malloc_set_alignment(32, mca_btl_openib_module.super.btl_eager_limit);
```

Comment:
This is a huge abstraction violation: you cannot have one component directly call another component.